common/local_utils.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Helper functions for using the local_utils tool."""
+
+import os
+
+from common import logs
+from common import new_process
+
+logger = logs.Logger('local_utils')
+
+
+def local_utils_command(arguments, *args, **kwargs):
+    """Executes a local_utils command with |arguments| and returns the result.
+    """
+    return new_process.execute(arguments, *args, **kwargs)
+
+
+def cp(*cp_arguments, **kwargs):  # pylint: disable=invalid-name
+    """Executes local_utils's "cp" command with |cp_arguments| and returns the
+    returncode and the output."""
+    # Create intermediate folders for `cp` command.
+    for file_or_dir_path in cp_arguments:
+        dirpath = os.path.dirname(os.path.abspath(file_or_dir_path))
+        if not os.path.exists(dirpath):
+            os.makedirs(dirpath)
+
+    command = ['cp']
+    command.extend(cp_arguments)
+    return local_utils_command(command, **kwargs)
+
+
+def ls(*ls_arguments, must_exist=True, **kwargs):  # pylint: disable=invalid-name
+    """Executes local_utils's "ls" command with |ls_arguments| and returns the
```

Comment:
I think this function is broken. The implementation from `gsutil` assumes results are printed line by line but this doesn't happen with just `ls` unless `-l` is used.
Luckily I think the only place this is used is in measurer.py's `remote_dir_exists`.
Maybe delete `ls`, move `remote_dir_exists` to `filestore_utils`, rename to `path_exists`. The implementations can call `ls` and `gsutil ls` and return True if returncode is zero otherwise False.

Reply:
Done.